lock/lock_test.go

```diff
@@ -0,0 +1,46 @@
+package lock
+
+import (
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/require"
+	"testing"
+	"time"
+
+	"github.com/gruntwork-io/go-commons/logging"
+	"github.com/stretchr/testify/assert"
+)
+
+var options = Options {
+	logging.GetLogger("test"),
+	"test-dynamodb-lock-string-" + random.UniqueId(),
+	"test-dynamodb-lock-table",
+	"eu-central-1",
+	2,
+	1 * time.Second,
+	true,
+}
+
+func TestAcquireLockWithRetries(t *testing.T) {
+	t.Parallel()
+
+	defer assertLockReleased(t, &options)
+	defer ReleaseLock(&options)
+
+	options.Logger.Infof("Acquiring first lock...\n")
+	err := AcquireLock(&options)
+	require.NoError(t, err)
+
+	options.Logger.Infof("Acquiring second lock...\n")
+	err = AcquireLock(&options)
+
+	if err == nil {
+		require.Error(t, err, "Acquiring of second lock succeeded, but it was expected to fail.\n")
+	}
```

Comment:
This is a good place to use [`assert.Error`](https://pkg.go.dev/github.com/stretchr/testify/assert#Error) to make the code slightly more legible. Note that the trailing newline [is automatically appended](https://github.com/stretchr/testify/blob/acba37e5db06f0093b465a7d47822bf13644b66c/assert/assertions.go#L262) and the `.` is kind of optional IMO for test error messages.
```suggestion
	assert.Error(t, err, "Acquiring of second lock succeeded, but it was expected to fail")
```

Reply:
Oh, that's actually how it was before. I believe there was a brief comment from Jim here that it might be easier to compare the error message and use the `require`. I've looked into what happens under the hood for the 2 methods, and the outcome for this case seems to be the same, although under the hood they'll call into different underlying functions. 

I'm happy to proceed with it being assert

Reply:
@brikis98 not super serious, but curious to see what you make of this - is there a compelling reason why you thought the `require` is a better choice here?

Reply:
There's no need for the `if err == nil` check. It's _just_ the one-liner with `require.xxx` or `assert.xxx`. Whether you use `require` or `assert` depends on whether the test should keep running or not. But the point is to replace a 3-line `if err = nil { ... }` block with a one-liner (which also has better error reporting).

Reply:
Ahh! Of course :) Noted 👍